### PR TITLE
[Do not merge] Use capnp in Libp2p helper

### PR DESCRIPTION
**Pull request by georgeee**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

Explain your changes here.

Explain how you tested your changes here.

Checklist:

- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

File: src/app/libp2p_helper/src/libp2p_helper/app.go [hunk 1]
```diff
@@ -1090,3 +1310,114 @@ func findPeerInfo(app *app, id peer.ID) (*codaPeerInfo, error) {
 	}
 	return maybePeer, nil
 }
+
+func (app *app) handleIncomingMsg(msg *ipc.Libp2pHelperInterface_Message) {
+	if msg.HasRpcRequest() {
```
Comment (nholland94) [inline src/app/libp2p_helper/src/libp2p_helper/app.go]: Can't we use the `msg.Which()` pattern for these unions instead? Then we can clean all of this up by using a jump table to lookup the handlers we want to dispatch to. For example:

```go
func (app *app) handleResetStream(seqno uint64, req ipc.Libp2pHelperInterface_RpcRequest) *capnp.Message {
  m, err := req.ResetStream()
  panicOnErr(err)
  ...
}

type rpcRequestHandler = func(uint64, ipc.Libp2pHelperInterface_RpcRequest) *capnp.Message

var rpcRequestHandlers : map[ipc.Libp2pHelperInterface_RpcRequest_Which]rpcRequestHandler = {
  ipc.Libp2pHelperInterface_RpcRequest_Which_resetStream: handleResetStream,
  ...
}
```

**Review by nholland94 — CHANGES_REQUESTED** [review]
Looks good to me overall. Noticed a small issue with a log statement, and have a suggestion to clean up the entrypoint for handling incoming RPC messages.

File: src/app/libp2p_helper/src/libp2p_helper/app.go [hunk 2]
```diff
@@ -284,57 +287,115 @@ func (app *app) checkLatency() {
...
+		default:
+			app.P2p.Logger.Warningf("handleValidation: unknown validation result %d", m.Result())
+		}
+		st.Completion <- res
+		if st.TimedOutAt != nil {
+			app.P2p.Logger.Errorf("validation for item %d took %d seconds", seqno, time.Now().Add(validationTimeout).Sub(*st.TimedOutAt))
+		}
+		delete(app.Validators, seqno)
+	}
+	app.P2p.Logger.Warningf("handleValidation: validation seqno %d unknown", seqno)
```
[the remark this answers is not in the document]
Comment (georgeee) [reply src/app/libp2p_helper/src/libp2p_helper/app.go]: Fixed

File: src/app/libp2p_helper/src/libp2p_helper/app.go (hunk 1 above)
Comment (georgeee) [reply src/app/libp2p_helper/src/libp2p_helper/app.go]: Implemented the change, good idea!

**Comment by georgeee** [comment]
Changes are to be delivered to `compatible` via https://github.com/MinaProtocol/mina/pull/9504

**Pull request closed without merging**